compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -148,7 +148,12 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisPhase
           checkInferredWellFormed(tree.tpt)
           val sym = tree.symbol
           if sym.isScala2Macro && !ctx.settings.XignoreScala2Macros.value then
-            if !sym.owner.unforcedDecls.exists(p => !p.isScala2Macro && p.name == sym.name && p.signature == sym.signature) then
+            if !sym.owner.unforcedDecls.exists(p => !p.isScala2Macro && p.name == sym.name && p.signature == sym.signature)
+               // Allow scala.reflect.materializeClassTag to be able to compile scala/reflect/package.scala
+               // At this point we cannot force the symbol of scala.reflect.package$, hence we check it by name
+               // This should be removed on Scala 3.1
+               && sym.showFullName != "scala.reflect.materializeClassTag"
```

Comment:
We shouldn't rely on the pretty-printer for this. You can use `fullName.toString` instead, but I don't understand why you can't force the symbol, you shouldn't get cycles in PostTyper.

Reply:
Removed from this PR

Reply:
Added it again

Reply:
But now I check the symbol